### PR TITLE
#917 outbound shipment navigating with number

### LINKED ISSUE #917: shipments should use shipment numbers for queries not IDs
Currently using `id` for navigating to detail views. Should use `shipmentNumber`

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #917

- Same as the other one - went offroad moving all the API code for outbounds this time
- Added navigating by invoice numbers rather than IDs

### CODE REVIEW

File: packages/invoices/src/OutboundShipment/api/hooks.ts [hunk 1]
```diff
@@ -18,48 +22,142 @@ import {
...
+  const navigate = useNavigate();
+  const api = useOutboundApi();
+  return useMutation(api.insert, {
+    onSuccess: invoiceNumber => {
+      const route = RouteBuilder.create(AppRoute.Distribution)
+        .addPart(AppRoute.OutboundShipment)
+        .addPart(String(invoiceNumber))
+        .build();
+      navigate(route, { replace: true });
+      queryClient.invalidateQueries(['invoice']);
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/hooks.ts:103]: using a full path here so that the hook can be reused in the dashboard etc

**Review by joshxg — COMMENTED** [review]
Don't have much commentary to add that could be helpful, sorry.. just a bunch of shiftin' stuff.

File: packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx [hunk 2]
```diff
@@ -1,69 +1,49 @@
...
-  useListData,
   useNotification,
   useQuery,
   useTranslation,
   StatsPanel,
   Widget,
-  useNavigate,
-  useQueryParams,
+  generateUUID,
+  useToggle,
```
Comment (mark-prins) [inline packages/invoices/src/OutboundShipment/Dashboard/Widget.tsx:13]: very nice! good use of the useToggle and useCreateOutbound hooks

**Review by mark-prins — APPROVED** [review]
👍  LGTM!

**Pull request merged**